[Hub Generated] Review request for Microsoft.VMwareCloudSimple to add version stable/2019-04-01

tenmozes opened the pull request:
If you are a MSFT employee you can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/tenmozes/azure-rest-api-specs/dev-vmwarecloudsimple-Microsoft.VMwareCloudSimple-2019-04-01...master).

### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

tenmozes committed:
New Readme Config File

tenmozes committed:
New Go Language Readme Config File

tenmozes committed:
New Typescript Language Readme Config File

tenmozes committed:
New Python Language Readme Config File

tenmozes committed:
New C# Language Readme Config File

tenmozes committed:
New Ruby Language Readme Config File

tenmozes committed:
New Swagger Spec File

tenmozes committed:
New Swagger Example Spec File

Commit:
[VMwareCloudSimple] initial spec

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/Azure/azure-rest-api-specs?pullRequest=6196) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/Azure/azure-rest-api-specs?pullRequest=6196) before we can accept your contribution.<br/><br/>:x: tenmozes [sign now](https://cla.opensource.microsoft.com/Azure/azure-rest-api-specs?pullRequest=6196)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/Azure/azure-rest-api-specs?pullRequest=6196) it.</sub>

AutorestCI commented:
# Automation for azure-sdk-for-python

AutorestCI commented:
# Automation for azure-sdk-for-js

AutorestCI commented:
# Automation for azure-sdk-for-java

AutorestCI commented:
# Automation for azure-sdk-for-ruby

AutorestCI commented:
# Automation for azure-sdk-for-go

azuresdkci commented:
Can one of the admins verify this patch?

Commit:
[VMwareCloudSimple] clean-up examples, spellcheck

tenmozes committed:
[VMwareCloudSimple] return 202 for long-running vm stop/start

ravbhatnagar commented on specification/vmwarecloudsimple/resource-manager/Microsoft.VMwareCloudSimple/stable/2019-04-01/vmwarecloudsimple.json:
```diff
@@ -0,0 +1,3604 @@
+              "$ref": "#/definitions/DedicatedCloudNodeListResponse"
+            }
+          },
+          "404": {
```
you dont need to list the 404 response separately. It can be covered via the default response.

ravbhatnagar commented on specification/vmwarecloudsimple/resource-manager/Microsoft.VMwareCloudSimple/stable/2019-04-01/vmwarecloudsimple.json:
```diff
@@ -0,0 +1,3604 @@
+      "properties": {
+        "serviceSpecification": {
+          "description": "The list of specification's service metrics",
+          "$ref":
```
There is a standard contract for surfacing/targeting AZs in Azure. Customers should not confuse this with that. May be we change the property name to add a prefix to indicate this is VmWare specific? Lets sync with Amit on this one.

ei82 replied:
@ravbhatnagar How critical the requested change is? 
The problem is that CloudSimple RP is already in production and resolution of the comment is going to take a while.  
Can we address it in one of the following smaller pull requests?

ravbhatnagar replied:
ok lets take it up later, but please take a note of this.

ravbhatnagar commented on specification/vmwarecloudsimple/resource-manager/Microsoft.VMwareCloudSimple/stable/2019-04-01/vmwarecloudsimple.json:
```diff
@@ -0,0 +1,3604 @@
+      "properties": {
+        "serviceSpecification": {
+          "description": "The list of specification's service metrics",
+          "$ref":
```
read only

ravbhatnagar replied:
is this needed?

ravbhatnagar replied:
Move API need not be described in RP swagger. Looks like the request body wasnt removed.

ravbhatnagar commented on specification/vmwarecloudsimple/resource-manager/Microsoft.VMwareCloudSimple/stable/2019-04-01/vmwarecloudsimple.json:
```diff
@@ -0,0 +1,3604 @@
+      "properties": {
+        "serviceSpecification": {
+          "description": "The list of specification's service metrics",
+          "$ref":
```
how do they get this?

tenmozes replied:
'/subscriptions/{subscriptionId}/providers/microsoft.vmwarecloudsimple/locations/{regionId}/availabilities':  endpoints returns information how many nodes are available for particular region, availability zone and placement group

ravbhatnagar commented on specification/vmwarecloudsimple/resource-manager/Microsoft.VMwareCloudSimple/stable/2019-04-01/vmwarecloudsimple.json:
```diff
@@ -0,0 +1,3604 @@
+      "properties": {
+        "serviceSpecification": {
+          "description": "The list of specification's service metrics",
+          "$ref":
```
Cant this be surfaced in the sku top level property?

tenmozes replied:
This model a bit different, it contains information mostly for displaying on UI however sku property on a first level has all necessary information in a standard format

ravbhatnagar commented on specification/vmwarecloudsimple/resource-manager/Microsoft.VMwareCloudSimple/stable/2019-04-01/vmwarecloudsimple.json:
```diff
@@ -0,0 +1,3604 @@
+      "properties": {
+        "serviceSpecification": {
+          "description": "The list of specification's service metrics",
+          "$ref":
```
may be rename the property to accountOnboardingState

tenmozes replied:
will address in next PR

ravbhatnagar reviewed (commented):
@tenmozes - few comments. please take a look.

tenmozes committed:
[VMwareCloudSimple] address PR comments
 - change case of paths
 - add readOnly to created property
 - remove 404
 - remove MoveBody model

ravbhatnagar reviewed (commented):
signing off from ARM side

daschult commented on specification/vmwarecloudsimple/resource-manager/readme.md:
```diff
@@ -0,0 +1,79 @@
+  - repo: azure-sdk-for-js
+  - repo: azure-sdk-for-ruby
+    after_scripts:
+      - bundle install && rake arm:regen_all_profiles['azure_mgmt_vmwarecloudsimple']
```
Please remove this `after_scripts` section (both lines 57 and 58).

daschult commented on specification/vmwarecloudsimple/resource-manager/readme.md:
````diff
@@ -0,0 +1,79 @@
+
+```yaml $(swagger-to-sdk)
+swagger-to-sdk:
+  - repo: azure-sdk-for-python
````
Please add a `- repo: azure-sdk-for-net` section here, too.

daschult commented on specification/vmwarecloudsimple/resource-manager/readme.typescript.md:
````diff
@@ -0,0 +1,13 @@
+```yaml $(typescript)
+typescript:
+  azure-arm: true
+  package-name: "vmwarecloudsimple"
````
This appears to be a management-plane package, so please prefix the package name with `@azure/arm-`.
```suggestion
  package-name: "@azure/arm-vmwarecloudsimple"
```

daschult commented on specification/vmwarecloudsimple/resource-manager/readme.typescript.md:
```diff
@@ -0,0 +1,13 @@
+typescript:
+  azure-arm: true
+  package-name: "vmwarecloudsimple"
+  output-folder: "$(typescript-sdks-folder)/packages/vmwarecloudsimple"
```
Update the output folder to follow the new folder structure in azure-sdk-for-js:
```suggestion
  output-folder: "$(typescript-sdks-folder)/sdk/vmwarecloudsimple/arm-vmwarecloudsimple"
```

amarzavery commented:
@sarangan12 , @shahabhijeet @jhendrixMSFT , @lmazuel, @anuchandy  - Please take a look at the readme.md and readme.lang.md files and make sure the settings are correct.

tenmozes committed:
update readme.me and readme.typescript.md. Fix output path, add azure-sdk-for-net

jhendrixMSFT commented on specification/vmwarecloudsimple/resource-manager/readme.go.md:
```diff
@@ -0,0 +1,19 @@
+## Go
```
File is incorrect, please see https://github.com/Azure/azure-rest-api-specs/blob/master/specification/batch/resource-manager/readme.go.md for a template.

jhendrixMSFT reviewed (changes requested):
readme.go.md needs  fixing

tenmozes committed:
update readme.go.md

jhendrixMSFT commented on specification/vmwarecloudsimple/resource-manager/readme.go.md:
````diff
@@ -0,0 +1,19 @@
+Please also specify `--go-sdks-folder=<path to the root directory of your azure-sdk-for-go clone>`.
+
+```yaml $(tag) == 'package-2019-04-01' && $(go)
+output-folder: $(go-sdks-folder)/services/$(namespace)/mgmt/2019-04-01/$(namespace)
````
Extra 's', should be `$(go-sdk-folder)`

jhendrixMSFT commented on specification/vmwarecloudsimple/resource-manager/readme.go.md:
````diff
@@ -0,0 +1,19 @@
+  namespace: vmwarecloudsimple
+  clear-output-folder: true
+```
+
````
Missing section `Go multi-api`

tenmozes replied:
@jhendrixMSFT thanks for review, comments are addressed

jhendrixMSFT reviewed (approved):
readme.go.md looks good

daschult commented:
@openapi sdkautomation rebuild

daschult commented:
@openapi sdkautomation rebuild

kpajdzik commented:
@openapi sdkautomation rebuild

daschult commented:
@openapi sdkautomation rebuild

tenmozes committed:
update readme.python.md

tenmozes committed:
update readme.ruby.md

tenmozes committed:
update readme.csharp.md

tenmozes committed:
readme.ruby.md wrap namespace with quotes

anuchandy commented on specification/vmwarecloudsimple/resource-manager/Microsoft.VMwareCloudSimple/stable/2019-04-01/vmwarecloudsimple.json:
```diff
@@ -0,0 +1,3470 @@
+            "$ref": "./examples/GetOperationResult.json"
+          }
+        },
+        "x-ms-long-running-operation": true
```
This is a GET operation to retrieve operation status of LRO, ```x-ms-long-running-operation``` annotation is not valid for this operation, please remove this annotation.

anuchandy replied:
java codegen has a check for incorrect use of annotations, so after removing this annotation, please merge my PR that adds java tag https://github.com/tenmozes/azure-rest-api-specs/pull/1

tenmozes replied:
@anuchandy thank you for a review, fixed

tenmozes committed:
remove x-ms-log-running-operation from get

tenmozes committed:
happy spellchecker

nschonni commented on custom-words.txt:
```diff
@@ -908,13 +913,16 @@ overridable
 OWASP
 oxxm
 pageable
+PARALEL
```
Is this not a typo?

tenmozes replied:
thanks fixed, that was a typo

tenmozes committed:
happy spellchecker. fix typos

tenmozes committed:
update readme.python.md

tenmozes committed:
add readme.java.md

tenmozes committed:
update readme.python.md change output-folder
